resources/splunk_service.rb

```diff
@@ -114,10 +131,10 @@ class LinuxService < SplunkService
 
     if changed? :ulimit
       write_initd_ulimit ulimit
-      ensure_restart if service_running
+      run_context.notifies_immediately(Notification.new(current_resource, :desired_restart, current_resource)) if service_running
     end
 
-    service_action :start
+    service_action :start unless changed?(:ulimit) && service_running
```

Comment:
I'm not sure if you need the guards here and on line 134... but there might be an interesting ordering thing between the immediate notification and the guards that you may want to check out.